Detectors/TRD/workflow/src/TRDTrapSimulatorSpec.cxx

```diff
@@ -221,25 +223,30 @@ void TRDDPLTrapSimulatorTask::run(o2::framework::ProcessingContext& pc)
 
   // input
   auto inputDigits = pc.inputs().get<gsl::span<o2::trd::Digit>>("digitinput");                                 // block of TRD digits
-  auto digitMCLabels = pc.inputs().get<o2::dataformats::ConstMCTruthContainer<o2::MCCompLabel>>("labelinput"); // MC labels associated to the input digits
   auto inputTriggerRecords = pc.inputs().get<gsl::span<o2::trd::TriggerRecord>>("triggerrecords");             // time and number of digits for each collision
   if (inputDigits.size() == 0 || inputTriggerRecords.size() == 0) {
     LOG(warn) << "Did not receive any digits, trigger records, or neither one nor the other. Aborting.";
     return;
   }
   LOG(debug) << "Read in " << inputDigits.size() << " digits";
-  LOG(debug) << "Labels contain " << digitMCLabels.getNElements() << " elements with and indexed size of " << digitMCLabels.getIndexedSize();
+
+  const o2::dataformats::ConstMCTruthContainer<o2::MCCompLabel>* lblDigitsPtr = nullptr;
+
+//  if (mUseMC) {
```

Comment:
The problem is here, `auto lblDigits` goes out of scope after the `if (museMC) { ... }` branch, so when lblDigitsPtr is used below, it is invalid.

Reply:
Thank you very much, that makes perfectly sense. I just didn't see it! After the compilation is done and I have tested it I will push the changes.